engine/src/operators/OrderBy.cpp

```diff
@@ -321,9 +321,187 @@ void process_sort(blazing_frame & input, std::string query_part, Context * query
 	}
 }
 
+
+
+}  // namespace operators
+}  // namespace ral
+
+namespace ral {
+namespace operators {
+namespace experimental {
+
+	using Context = blazingdb::manager::experimental::Context;
+	using Node = blazingdb::transport::experimental::Node;
+	using namespace ral::distribution::experimental;
+
+std::unique_ptr<ral::frame::BlazingTable> process_sort(const ral::frame::BlazingTableView & table, std::string query_part, Context & context) {
+	auto rangeStart = query_part.find("(");
+	auto rangeEnd = query_part.rfind(")") - rangeStart - 1;
+	std::string combined_expression = query_part.substr(rangeStart + 1, rangeEnd);
+
+	std::string limitRowsStr = get_named_expression(combined_expression, "fetch");
+
+	size_t num_sort_columns = count_string_occurrence(combined_expression, "sort");
+
+	std::vector<int8_t> sortOrderTypes(num_sort_columns);
+	std::vector<int> sortColIndices(num_sort_columns);
+	for(int i = 0; i < num_sort_columns; i++) {
+		sortColIndices[i] = get_index(get_named_expression(combined_expression, "sort" + std::to_string(i)));
+		sortOrderTypes[i] =
+			(get_named_expression(combined_expression, "dir" + std::to_string(i)) == DESCENDING_ORDER_SORT_TEXT);		 
+	}
+
+	if(context.getTotalNodes() <= 1) {
```

Comment:
dont make seperate logic for when its not distributed. If theres only 1 node you can have the functions basically return their inputs using hte pattern we discussed so that everything follows the same flow

Reply:
what pattern are you talking about? If you want just one function that handles distributed or not distributed, use this ```std::unique_ptr<ral::frame::BlazingTable> process_sort(const ral::frame::BlazingTableView & table, const std::string & query_part, Context * context)```

Reply:
What I am talking about is not making two longs paths of execution that diverge when they dont need to. The fewer paths of divergence we have the less code we write

Reply:
But if we just have it as one path, then we have extra steps we dont want to be taking. For example, generating samples, sending samples, collecting samples, partitionning data and so on.  Also, if a bigger than GPU context, its my understanding that some of these distirbution primitives would be getting done in separate machines, so why do we care if its all in one flow now? if that is not how we would be doing it in bigger than GPU anyways

Reply:
If those functions ndo nothing when they dont need to do anything because they only have one node they should do what we described about telling it to use columns from teh input table and pass it to the output to avoid copying, this way we onloy hvae one flow

distribute partitions would do nothing, collect partitions nwould just return the thing you started with without making a copy. This way we get rid of multiple paths of execution

Reply:
So generate samples would have, `if(context.getTotalNodes() <= 1) {` and distributed samples would have its own if, and collect samples would have `if(context.getTotalNodes() <= 1) {` .... that seems like we are still having a diverging path, except now its all over the place

Reply:
Also, please answer my question, how are those distribution primitives going to work in bigger than GPU? Arent they going to be in diferent machines? if so, then why does it matter right now? If what you need is for the distribution primitives to work in  a certain way, then that is a different matter. You need to specify exactly what those needs are